smt.go

```diff
@@ -451,7 +460,10 @@ func (smt *SMT) ProveClosest(path []byte) (
 				node = ext.expand()
 			}
 		}
-		inner := node.(*innerNode)
+		inner, ok := node.(*innerNode)
+		if !ok { // this can only happen for an empty tree
+			break
```

Comment:
add a print statement!

Reply:
@Olshansk Mentioned this below but I think the result of the return values should be used instead of a print. Besides debugging the SMT itself is there a usecase for print statements with the new return values clearly showing the tree was empty vs. 1 node vs N nodes

Reply:
Fair enough. Resolving